# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: ipc/glue/ProtocolUtils.h
# Commit: 7b7819250f79
# Full Hash: 7b7819250f7965227783233af6de0e4e1b789af6
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2021-06-23 09:53:24
# Regressor Bug: 1706374
# File Overlap Count: 1
# Description:
#   Bug 1706374 - Part 11: Add NodeController component bridging IPC and Ports, r=handyman
#   
#   The NodeController and NodeChannel types act as the backbone connecting the
#   existing IPC logic and driving the ports routing code. Individual NodeChannel
#   objects wrap and respond to messages from IPC::Channel, and the NodeController
# ==============================================================================

diff -r 8bdac5eedf78 -r 7b7819250f79 ipc/glue/ProtocolUtils.h
--- a/ipc/glue/ProtocolUtils.h	Tue Jun 22 18:17:21 2021 +0000
+++ b/ipc/glue/ProtocolUtils.h	Tue Jun 22 18:17:22 2021 +0000
@@ -14,6 +14,7 @@
 #include "base/basictypes.h"
 #include "base/process.h"
 #include "chrome/common/ipc_message.h"
+#include "mojo/core/ports/port_ref.h"
 #include "mozilla/AlreadyAddRefed.h"
 #include "mozilla/Assertions.h"
 #include "mozilla/Attributes.h"
@@ -57,6 +58,14 @@
 // protocol 0.  Oops!  We can get away with this until protocol 0
 // starts approaching its 65,536th message.
 enum {
+  // Message types used by NodeChannel
+  ACCEPT_INVITE_MESSAGE_TYPE = kuint16max - 14,
+  REQUEST_INTRODUCTION_MESSAGE_TYPE = kuint16max - 13,
+  INTRODUCE_MESSAGE_TYPE = kuint16max - 12,
+  BROADCAST_MESSAGE_TYPE = kuint16max - 11,
+  EVENT_MESSAGE_TYPE = kuint16max - 10,
+
+  // Message types used by MessageChannel
   IMPENDING_SHUTDOWN_MESSAGE_TYPE = kuint16max - 9,
   BUILD_IDS_MATCH_MESSAGE_TYPE = kuint16max - 8,
   BUILD_ID_MESSAGE_TYPE = kuint16max - 7,  // unused
@@ -435,6 +444,8 @@
   virtual void OnChannelError() = 0;
   virtual void ProcessingError(Result aError, const char* aMsgName) {}
 
+  bool Open(ScopedPort aPort, base::ProcessId aOtherPid);
+
   bool Open(UniquePtr<Transport> aTransport, base::ProcessId aOtherPid,
             MessageLoop* aThread = nullptr,
             mozilla::ipc::Side aSide = mozilla::ipc::UnknownSide);